staging/src/k8s.io/api/apps/v1/types.go

```diff
@@ -116,6 +116,40 @@ type RollingUpdateStatefulSetStrategy struct {
 	Partition *int32 `json:"partition,omitempty" protobuf:"varint,1,opt,name=partition"`
 }
 
+// PersistentVolumeClaimRetentionPolicyType is a string enumeration of the policies that will determine
+// when volumes from the VolumeClaimTemplates will be deleted when the controlling StatefulSet is
+// deleted or scaled down.
+type PersistentVolumeClaimRetentionPolicyType string
+
+const (
+	// RetainPersistentVolumeClaimRetentionPolicyType is the default
+	// PersistentVolumeClaimRetentionPolicy and specifies that
+	// PersistentVolumeClaims associated with StatefulSet VolumeClaimTemplates
+	// will not be deleted.
+	RetainPersistentVolumeClaimRetentionPolicyType PersistentVolumeClaimRetentionPolicyType = "Retain"
+	// RetentionPersistentVolumeClaimRetentionPolicyType specifies that
+	// PersistentVolumeClaims associated with StatefulSet VolumeClaimTemplates
+	// will be deleted in the scenario specified in
+	// StatefulSetPersistentVolumeClaimRetentionPolicy.
+	DeletePersistentVolumeClaimRetentionPolicyType PersistentVolumeClaimRetentionPolicyType = "Delete"
+)
+
+// StatefulSetPersistentVolumeClaimRetentionPolicy describes the policy used for PVCs
+// created from the StatefulSet VolumeClaimTemplates.
+type StatefulSetPersistentVolumeClaimRetentionPolicy struct {
+	// OnSetDeletion specifies what happens to PVCs created from StatefulSet
```

Comment:
Be sure to fix up these comments to match the new field name.

Reply:
done

Reply:
> /approve
> 
> but needs a few comment fixups

Thanks for catching those comment errors!

I don't want to change anything at this point, but was it actually necessary for me to update the v1beta1 and v1beta2 apis? That was my interpretation of https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api_changes.md but after having gone through all the changes & understanding k8s api implementation better I'm not sure I needed to have done that?

Reply:
Not absolutely necessary to add the fields in older versions, but doesn't hurt.

Reply:
Ah, thanks. Good to know. The extra toil I think ends up being significant.